packages/zoe/src/zoeService/utils.d.ts

```diff
@@ -38,18 +39,18 @@ export type InstallationStart<I> = I extends Installation<infer SF>
   ? SF
   : never;
 
-type StartParams<SF> = SF extends (
-  zcf: { getTerms: () => {} },
-  privateArgs: {},
-  baggage?: unknown,
-) => unknown
+type ContractStartFunction = (
+  zcf?: ZCF,
+  privateArgs?: {},
+  baggage?: Baggage,
+) => ERef<{ creatorFacet?: {}; publicFacet?: {} }>;
+
+type StartParams<SF> = SF extends ContractStartFunction
   ? {
       terms: ReturnType<Parameters<SF>[0]['getTerms']>;
       privateArgs: Parameters<SF>[1];
     }
-  : SF extends (zcf: { getTerms: () => {} }) => unknown
-  ? { terms: any }
-  : {};
+  : never;
 
 type StartResult<S> = S extends (...args: any) => Promise<infer U>
```

Comment:
`infer U` is magic I have yet to learn. That makes this an "I don't see any critical problems" review and not a "I'm prepared to maintain this code" review.